pkg/solana/fees/block_history_test.go

```diff
@@ -11,86 +12,366 @@ import (
+		rw.On("GetLatestBlock", mock.Anything).Return(nil, fmt.Errorf("fail rpc call")) // Mock GetLatestBlock returning error
+		estimator := initializeEstimator(tests.Context(t), t, rwLoader, cfg, logger.Test(t))
+
+		// Ensure the price remains unchanged
+		require.Error(t, estimator.calculatePrice(tests.Context(t)), "Expected error when GetLatestBlock fails")
+		cfg.On("ComputeUnitPriceMax").Return(max)
+		assert.Equal(t, uint64(100), estimator.BaseComputeUnitPrice(), "Price should not change when GetLatestBlock fails")
+	})
+
+	t.Run("Failed to Parse Block", func(t *testing.T) {
+		// Setup
+		rw := clientmock.NewReaderWriter(t)
+		rwLoader := utils.NewLazyLoad(func() (client.ReaderWriter, error) {
+			return rw, nil
+		})
+		cfg := cfgmock.NewConfig(t)
+		setupConfigMock(cfg, defaultPrice, min, max, pollPeriod, depth)
+		rw.On("GetLatestBlock", mock.Anything).Return(nil, nil) // Mock GetLatestBlock returning nil
+		estimator := initializeEstimator(tests.Context(t), t, rwLoader, cfg, logger.Test(t))
+
+		// Ensure the price remains unchanged
+		require.Error(t, estimator.calculatePrice(tests.Context(t)), "Expected error when parsing fails")
+		cfg.On("ComputeUnitPriceMax").Return(max)
+		assert.Equal(t, uint64(100), estimator.BaseComputeUnitPrice(), "Price should not change when parsing fails")
+	})
+
+	t.Run("Failed to Calculate Median", func(t *testing.T) {
+		// Setup
+		rw := clientmock.NewReaderWriter(t)
+		rwLoader := utils.NewLazyLoad(func() (client.ReaderWriter, error) {
+			return rw, nil
+		})
+		cfg := cfgmock.NewConfig(t)
+		setupConfigMock(cfg, defaultPrice, min, max, pollPeriod, depth)
+		rw.On("GetLatestBlock", mock.Anything).Return(&rpc.GetBlockResult{}, nil) // Mock GetLatestBlock returning empty block
+		estimator := initializeEstimator(tests.Context(t), t, rwLoader, cfg, logger.Test(t))
+
+		// Ensure the price remains unchanged
+		require.Error(t, estimator.calculatePrice(tests.Context(t)), "Expected error when median calculation fails")
+		cfg.On("ComputeUnitPriceMax").Return(max)
+		assert.Equal(t, uint64(100), estimator.BaseComputeUnitPrice(), "Price should not change when median calculation fails")
+	})
+
+	t.Run("Failed to Get Client", func(t *testing.T) {
+		// Setup
+		rwFailLoader := utils.NewLazyLoad(func() (client.ReaderWriter, error) {
+			// Return error to simulate failure to get client
+			return nil, fmt.Errorf("fail client load")
+		})
+		cfg := cfgmock.NewConfig(t)
+		setupConfigMock(cfg, defaultPrice, min, max, pollPeriod, depth)
+		estimator := initializeEstimator(tests.Context(t), t, rwFailLoader, cfg, logger.Test(t))
+
+		// Call calculatePrice and expect an error
+		// Ensure the price remains unchanged
+		require.Error(t, estimator.calculatePrice(tests.Context(t)), "Expected error when getting client fails")
+		cfg.On("ComputeUnitPriceMax").Return(max)
+		assert.Equal(t, defaultPrice, estimator.BaseComputeUnitPrice(), "Price should remain at default when client fails")
+	})
```

Comment:
Could you add a similar test to the multiple block `no compute unit prices collected` for latest block tests? Looking at the current code, we're relying on the `mathutil.Median` to throw an error if the slice is empty but think it might be better to have an explicit check somewhere in between [here](https://github.com/smartcontractkit/chainlink-solana/blob/1cf95ab5742b081f1d35e6c7c7ba50fc02679851/pkg/solana/fees/block_history.go#L115-L124) with its own unique error string if no prices are returned.